mlflow/store/artifact/optimized_s3_artifact_repo.py

```diff
@@ -70,6 +72,7 @@ def _refresh_credentials(self):
         self._access_key_id = new_creds["access_key_id"]
         self._secret_access_key = new_creds["secret_access_key"]
         self._session_token = new_creds["session_token"]
+        self._s3_upload_extra_args = new_creds["s3_upload_extra_args"]
```

Comment:
Is this key guaranteed to be there?

Reply:
Yes: the credential refresh function found [here](https://github.com/mlflow/mlflow/pull/12495/files#diff-b8949a6cc9b6ca52729fc888a57fb444676a1f2751737b0a4e54a4b419fa762fR163-R171) ensures this key is present.